components/proxy/conf/Caddyfile

```diff
@@ -12,10 +12,10 @@
 
 	# configure plugin order
 	# https://caddyserver.com/docs/caddyfile/directives#directive-order
```

Comment:
I don't think we should change proxy. As far as I understand the ide proxy will have own subdomain on global load balancer, i.e. `ide.gitpod.io`.

Reply:
Passing the traffic through "proxy" can be nice, because it works without variance in all three scenarios: SaaS, preview envs, and gitpod self-hosted. Thus unless you insist on having a dedicated subdomain (e.g. ide.gitpod.io) I'm also in favour of adding the `/ide/` route to proxy.

Reply:
The idea is to have a dedicated IDE proxy for all publish IDE services, i.e. static files (local apps, images), blobserve, and OpenVSX proxy.

If we route it through meta proxy won't we compete with them for resources, i.e. connection pool and potentially degrade performance on one or another side? As I remember we already have such issues before.

Reply:
> If we route it through meta proxy won't we compete with them for resources, i.e. connection pool and potentially degrade performance on one or another side? As I remember we already have such issues before.

Up to the @gitpod-io/engineering-meta team :)
My guess is that the few images that will only be loaded when a user opens settings won't tip the scale.

If you want a subdomain via GLB in SaaS, how would the proxy be exposed in Gitpod self-hosted?

Reply:
> If you want a subdomain via GLB in SaaS, how would the proxy be exposed in Gitpod self-hosted?

@corneliusludmann as for now we use meta proxy as a fallback for openvsx proxy for self-hosted?

Reply:
Indeed. We do.

https://github.com/gitpod-io/gitpod/blob/e5be2b7225e06c68813a6dac86baff2a5f1d6091/chart/templates/proxy-configmap.yaml#L54-L65

Reply:
Let's wait on @csweichel take on it. But @meysholdt [explained](https://gitpod.slack.com/archives/C01KGM9EBD4/p1639059965130400?thread_ts=1639053623.123700&cid=C01KGM9EBD4) that we can follow the similar to OpenVSX proxy approach for now and later move OpenVSX proxy behind ide proxy, i.e `ide.gitpod.io/open-vsx`. 

@corneliusludmann Do you think there any draw back of running OpenVSX proxy not on the root path?

Reply:
@iQQBot Let's continue with `ide.gitpod.io`. We will deploy similarly how it is done for OpenVSX proxy right now.

Reply:
If you want more autonomy (because of load concerns, decployments, e.g.), the openvsx approach looks a nice trade-off between complexity/variance and decoupling.

In general, though, I see `proxy` less of an `meta`-exclusive thing, but more of a general ingress point to enable the "single-external-IP" scenarios (as @meysholdt pointed out). For the sake of simplicity - and unless there aren't any other concerns, see openvsx-proxy, - I see no problems with having all services belonging to the IDE team hanging behind the same `proxy`.